src/kernels/kernelConnector.ts

```diff
@@ -123,7 +123,7 @@ export class KernelConnector {
             clearInstalledIntoInterpreterMemento(memento, Product.ipykernel, metadata.interpreter.uri).ignoreErrors();
         }
 
-        const handleResult = await errorHandler.handleKernelError(
+        const handleResult = await errorHandler?.handleKernelError(
```

Comment:
Yes, not sure why this too is marked as nullable.

Reply:
Some comments would help if we don't want to change the types.

Reply:
we also don't seem to have a case in the switch statement below for handleResult being null - is that scenario actually supported?

Reply:
I was able to debug this with an invalid kernel. THe error handler is there. I'll remote the tryGet for error handler.